packages/mui-joy/src/styles/variantUtils.ts

```diff
@@ -4,57 +4,57 @@ import { ColorPaletteProp, DefaultColorPalette } from './types/colorSystem';
+  lightDisabledColor: `var(--joy-palette-${color}-300)`,
   lightDisabledBg: `var(--joy-palette-${color}-50)`,
 
-  containedColor: `#fff`,
-  containedBg: `var(--joy-palette-${color}-${color === 'neutral' ? '600' : '500'})`,
-  containedHoverBg: `var(--joy-palette-${color}-600)`,
-  containedActiveBg: `var(--joy-palette-${color}-400)`,
-  containedDisabledBg: `var(--joy-palette-${color}-300)`,
+  containedColor: '#fff',
+  containedBg: `var(--joy-palette-${color}-600)`,
+  containedHoverBg: `var(--joy-palette-${color}-700)`,
+  containedActiveBg: `var(--joy-palette-${color}-800)`,
+  containedDisabledBg: `var(--joy-palette-${color}-200)`,
 });
 
 export const createDarkModeVariantVariables = (color: ColorPaletteProp) => ({
-  textColor: `var(--joy-palette-${color}-200)`,
-  textHoverBg: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-800)`,
-  textActiveBg: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-700)`,
-  textDisabledColor: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-500)`,
+  textColor: `var(--joy-palette-${color}-300)`,
+  textHoverBg: `var(--joy-palette-${color}-800)`,
+  textActiveBg: `var(--joy-palette-${color}-700)`,
+  textDisabledColor: `var(--joy-palette-${color}-800)`,
 
   outlinedColor: `var(--joy-palette-${color}-200)`,
-  outlinedBorder: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-700)`,
-  outlinedHoverBg: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-800)`,
-  outlinedHoverBorder: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-600)`,
-  outlinedActiveBg: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-700)`,
-  outlinedDisabledColor: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-500)`,
-  outlinedDisabledBorder: `var(--joy-palette-${color === 'primary' ? 'neutral' : color}-800)`,
+  outlinedBorder: `var(--joy-palette-${color}-700)`,
+  outlinedHoverBg: `var(--joy-palette-${color}-900)`,
+  outlinedHoverBorder: `var(--joy-palette-${color}-600)`,
+  outlinedActiveBg: `var(--joy-palette-${color}-900)`,
+  outlinedDisabledColor: `var(--joy-palette-${color}-800)`,
+  outlinedDisabledBorder: `var(--joy-palette-${color}-800)`,
 
-  lightColor: `var(--joy-palette-${color}-300)`,
-  lightBg: `var(--joy-palette-${color}-800)`,
-  lightHoverBg: `var(--joy-palette-${color}-700)`,
-  lightActiveBg: `var(--joy-palette-${color}-600)`,
-  lightDisabledColor: `var(--joy-palette-${color}-500)`,
-  lightDisabledBg: `var(--joy-palette-${color}-800)`,
+  lightColor: `var(--joy-palette-${color}-200)`,
+  lightBg: `var(--joy-palette-${color}-900)`,
+  lightHoverBg: `var(--joy-palette-${color}-800)`,
+  lightActiveBg: `var(--joy-palette-${color}-700)`,
+  lightDisabledColor: `var(--joy-palette-${color}-800)`,
+  lightDisabledBg: `var(--joy-palette-${color}-900)`,
 
   containedColor: `#fff`,
-  containedBg: `var(--joy-palette-${color}-500)`,
+  containedBg: `var(--joy-palette-${color}-600)`,
```

Comment:
I'm wondering if it wouldn't be better to supply these values to the `background` property (instead of `background-color`). It would allow developers to override the theme with gradients (or even images, but I wouldn't consider it a popular use case).

Reply:
The idea of variant is to be applied globally (for most components). I think in case of gradient or image, it will be specific for component and as always developers can use the same theming API to do that. https://codesandbox.io/s/joy-cra-typescript-forked-be5uj?file=/src/App.tsx

However, if we see a real use case in the future we can also revisit this point.